src/annotator/components/NotebookModal.js

```diff
@@ -0,0 +1,96 @@
+import { createSidebarConfig } from '../config/sidebar';
+import { useEffect, useRef, useState } from 'preact/hooks';
+import classnames from 'classnames/bind';
+
+// FIXME: use the button from the frontend shared package once this is stable.
+import Button from '../../sidebar/components/Button';
+
+/**
+ *
+ * @typedef NotebookIframeProps
+ * @prop {Record<string, any>} config
+ * @prop {string} groupId
+ */
+
+/**
+ * Create the iframe that will load the notebook application.
+ * 
+ * @param {NotebookIframeProps} props
+
+ */
+function NotebookIframe({ config, groupId }) {
+  const notebookConfig = createSidebarConfig(config);
+  // Explicity set the "focused" group
+  notebookConfig.group = groupId;
+  const configParam = encodeURIComponent(JSON.stringify(notebookConfig));
+  const notebookAppSrc = `${config.notebookAppUrl}#config=${configParam}`;
+
+  return (
+    <iframe
+      title={'Hypothesis annotation notebook'}
+      className="Notebook__iframe"
+      // Enable media in annotations to be shown fullscreen
+      allowFullScreen
+      src={notebookAppSrc}
+    />
+  );
+}
+
+/**
+ * @typedef NotebookModalProps
+ * @prop {import('../../types/annotator').EventBus} eventBus
+ * @prop {Record<string, any>} config
+ */
+
+/**
+ *
+ * @param {NotebookModalProps} props
+ */
+export default function NotebookModal({ eventBus, config }) {
+  const [isHidden, setIsHidden] = useState(true);
+  const [groupId, setGroupId] = useState('');
+  const originalDocumentOverflowStyle = useRef('');
+  let emitter = useRef(
```

Comment:
Is there any reason `emitter` can't be const?

Reply:
Yes, there is a reason. 

Adding `const emitter = eventBus.createEmitter()` at line 53, makes the `useEffect` to depend on `emitter` instead on `eventBust`. On every state update, for example when a new `groupId` is emitted, the above line will run causing the emitter to point to a different object, which will run the cleanup function of the useEffect and cause the `document.body.style.overflow` to be reset.

It is possible to avoid the negative side effect and use `const emitter = eventBus.createEmitter()` by no adding `emitter` to the dependency list in `useEffect`, but I think it is not very clean.

Reply:
Sorry, I think I miss-interpreted your suggestion. Yes, I will make it a `const`.